src/components/Facility/StaffCapacity.tsx

```diff
@@ -82,150 +78,101 @@ export const StaffCapacity = (props: DoctorCapacityProps) => {
 
   const doctorTypes = getAllowedDoctorTypes(specializationsQuery.data?.results);
 
-  const isLastOptionType =
-    doctorTypes.filter((i) => i.disabled).length ===
-    DOCTOR_SPECIALIZATION.length - 1;
-
-  const headerText = !id ? "Add Staff Capacity" : "Edit Staff Capacity";
-  const buttonText = !id
-    ? `Save ${!isLastOptionType ? "& Add More" : "Staff Capacity"}`
-    : "Update Staff Capacity";
-
-  const validateData = () => {
+  const validateData = (form: typeof initForm, fieldName?: string) => {
     const errors = { ...initForm };
     let invalidForm = false;
-    Object.keys(state.form).forEach((field) => {
-      if (!state.form[field]) {
+    const fieldsToValidate = fieldName ? [fieldName] : Object.keys(form);
+    fieldsToValidate.forEach((field) => {
+      if (!form[field]) {
         errors[field] = t("field_required");
         invalidForm = true;
-      }
-      if (field === "count" && state.form[field] < 0) {
+      } else if (field === "count" && form[field] < 0) {
         errors[field] = "Staff count cannot be negative";
         invalidForm = true;
       }
     });
-    if (invalidForm) {
-      dispatch({ type: "set_error", errors });
-      return false;
-    }
     dispatch({ type: "set_error", errors });
-    return true;
+    return !invalidForm;
```

Comment:
Hmm, since we are negating it here and in handleSubmit for the check, let's convert to `validForm` for better readability, i.e. start with validForm as true, and then switch inside the conditionals and return that.

Reply:
Understood the changes and surely make them